src/signals/index.js

```diff
@@ -0,0 +1,19 @@
+import { useComputed, useSignal } from '@preact/signals'
+import { useEffect } from 'preact/hooks'
+
+import { defineStore } from "../core/defineStore.js";
+
+function useSyncExternalStoreSignal(subscribe, getSnapshot) {
```

Comment:
Why do we need a separated function? We can just call all `useEffect`/`useComputed` inside `useStore

Reply:
I am the uniformity of the "useStore*" implementation and the readability of the code. In this case, it is immediately clear that useStoreSignal has a similar implementation with useStore, but uses an alternative implementation of useSyncExternalStore.

Using a closure for useSyncExternalStoreSignal is not necessary. This function does not use variables from the scope of the useStoreSignal